src/layouts/EditHomepage.jsx

```diff
@@ -870,73 +859,60 @@ export default class EditHomepage extends Component {
         }
       });
     }
-    this.setState((currState) => ({
-      ...currState,
-      frontMatter: {
-        ...currState.frontMatter,
-        sections: newSections,
-      },
-      errors: newErrors,
-    }));
+    setFrontMatter({
+      ...frontMatter,
+      sections: newSections
+    })
+    setErrors(newErrors)
   }
 
-  toggleDropdown = async () => {
+  const toggleDropdown = async () => {
     try {
-      this.setState((currState) => ({
-        dropdownIsActive: !currState.dropdownIsActive,
-      }));
+      setDropdownIsActive(!dropdownIsActive)
```

Comment:
nit: state is not updated synchronously so to be safe we should access the previous hook state value using:
```
setDropdownIsActive((prevState) => !prevState)
```

Reply:
Good catch, resolved in https://github.com/isomerpages/isomercms-frontend/pull/334/commits/fba319b93f26be5223019069dcbc6a784a70bdc7